### PR TITLE
Sass templates

### LINKED ISSUE #22: Complete sass templates

### PR DESCRIPTION
**Pull request by ncoen-zaelot**
## PR Type

What kind of change does this PR introduce?

- [x] ✨ New feature
- [x] 💡 Code documentation

## Description

Issue: #22 

- Spacing
image

- Palette Main UI Colors
image

- Palette Color Coding
image

- Font Sizes
image

- Font Weights
image

## Does this PR introduce a breaking change?

- [x] Yes, it introduces a different behavior to an existing feature

Change scss variable names
---

## Does this PR introduce a change to the CSS?

- [c] Yes, it introduces a change, but the change will not affect any existing clients

---

## Reviewer's Scorecard

- [ ] Contains clear description of changes (2pts)
- [ ] Includes reproducible test instructions (2pts)
- [ ] 20 or less files affected (3pts)
- [ ] 10 or less commit statements (3pts)
- [ ] Relevant unit tests created or updated (5pts)
- [ ] Relevant E2E tests created or updated (5pts)

### CODE REVIEW

File: src/style/palette.scss [hunk 1]
```diff
@@ -1,17 +1,119 @@
...
+$ui-grey: #e5e5e5 !default;
+$riverstone: #f5f5f5 !default;
+$white: #ffffff !default;
 
-// primary
-$primary: $red;
-$primary-hover: $tomato;
-$primary-active: $orange;
+// main ui colors
+$primary: #f3e060 !default;
```
Comment (JavierZaelot) [inline src/style/palette.scss:41]: group this by type and not by state. For example `primary`, `primary-hover` and `primary-active` should be together.

File: src/style/palette.scss [hunk 2]
```diff
@@ -1,17 +1,119 @@
...
+
+$primary-active: lighten($primary, $color-modifier-percentage);
+$secondary-active: lighten($secondary, $color-modifier-percentage);
+$tertiary-active: lighten($tertiary, $color-modifier-percentage);
+$success-active: lighten($success, $color-modifier-percentage);
+$danger-active: lighten($danger, $color-modifier-percentage);
+$warning-active: lighten($warning, $color-modifier-percentage);
+
+// We export the variables to show the color codes in the story
+:export {
```
Comment (JavierZaelot) [inline src/style/palette.scss:65]: remove this export. Remember we don't want to export things that won't be used

File: src/style/palette.stories.jsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+  brown: 'Brown',
+  bubble: 'Bubble',
+  dark: 'Dark',
+  'wolf-grey': 'WolfGrey',
+  'ui-grey': 'UIGrey',
+  riverstone: 'Riverstone',
+  white: 'White'
+}
+
+const Color = ({ name, darkContrast, position }) => {
```
Comment (JavierZaelot) [inline src/style/palette.stories.jsx:65]: Why is this here?
Below you have the main colors and the color coding, not really sure why this is here

**Review by Javier-Miles — CHANGES_REQUESTED** [review]

File: src/style/palette.scss (hunk 1 above)
Comment (ncoen-zaelot) [reply src/style/palette.scss:41]: Sure 👍

File: src/style/palette.scss (hunk 2 above)
Comment (ncoen-zaelot) [reply src/style/palette.scss:65]: It is being used, to show the color codes in the stories

File: src/style/palette.stories.jsx (hunk 3 above)
Comment (ncoen-zaelot) [reply src/style/palette.stories.jsx:65]: That is the component that renders each color strip with it's name and color code, it's used below

File: src/style/palette.scss (hunk 2 above)
Comment (JavierZaelot) [reply src/style/palette.scss:68]: you should create classes for this. The same way you did for spacings and sizes

File: src/style/palette.stories.module.scss [hunk 4]
```diff
@@ -0,0 +1,450 @@
...
+  border-bottom-left-radius: 4px;
+}
+
+// Palette Color Classes
+
+
+.darkindigo {
+  background-color: palette.$darkindigo;
+}
+.darkindigo-code:after {
```
Comment (JavierZaelot) [inline src/style/palette.stories.module.scss:83]: don´t create a new css line for it, this is sass. It's magic

```
darkindigo {
  background-color: palette.$darkindigo;

  &:after {
    content: '#{palette.$darkindigo}';
  }
}
```

**Review by Javier-Miles — CHANGES_REQUESTED** [review]

File: src/style/palette.stories.jsx (hunk 3 above)
Comment (brunocerecetto) [reply src/style/palette.stories.jsx:61]: I don't think this is bad, but since the storybook is the documentation, I would tell you not to waste time doing components that are only used here

**Review by brunocerecetto — APPROVED** [review]

File: src/style/palette.stories.jsx (hunk 3 above)
Comment (ncoen-zaelot) [reply src/style/palette.stories.jsx:61]: I think I did it to **save** time, otherwise I would have repeated this 7 line component in each one of the 53 colors 😄 
I understand the point tho, thanks for the comments!
